sdk/storage/azure-storage-blob/azure/storage/blob/_shared/uploads_async.py

```diff
@@ -256,14 +258,15 @@ async def _upload_chunk(self, chunk_offset, chunk_data):
         await self.service.stage_block(
             block_id,
             len(chunk_data),
-            chunk_data,
+            body=chunk_data,
             data_stream_total=self.total_size,
             upload_stream_current=self.progress_total,
             **self.request_options)
         return index, block_id
 
-    async def _upload_substream_block(self, block_id, block_stream):
+    async def _upload_substream_block(self, index, block_stream):
         try:
+            block_id = 'BlockId{}'.format("%05d" % (index/self.chunk_size))
```

Comment:
oh you formatted the blockid here

Reply:
what will this generate?

Reply:
hahaha it's index! nvm

Reply:
why? index is i * self.chunk_size https://github.com/Azure/azure-sdk-for-python/pull/17272/files#diff-fd90426e10c45bf19310c14aef222c23145b307e761fef45eac642e704562103L224-L225

So (index/self.chunk_size) would be 0, 1, 2 etc